src/agent/src/config.rs

```diff
@@ -239,16 +239,22 @@ impl AgentConfig {
         let mut config: AgentConfig = Default::default();
         let cmdline = fs::read_to_string(file)?;
         let params: Vec<&str> = cmdline.split_ascii_whitespace().collect();
+
+        // Check if there is config file before parsing params that might
+        // override values from the config file.
         for param in params.iter() {
             // If we get a configuration file path from the command line, we
             // generate our config from it.
             // The agent will fail to start if the configuration file is not present,
             // or if it can't be parsed properly.
             if param.starts_with(format!("{}=", CONFIG_FILE).as_str()) {
                 let config_file = get_string_value(param)?;
-                return AgentConfig::from_config_file(&config_file);
+                config = AgentConfig::from_config_file(&config_file)?;
```

Comment:
Xynnn comment was that although we're doing this here, and it's fine, we're totally ignoring the fact that an agent config could be specified via command line arguments, and we'd still have to consider environment variables changing the config file.

With this in mind, we need to have an agreement on precedence.  If an agent config is passed via command-line, but **also** passed via environment variable, which one should be the one considered?  Whichever is chosen, we should warn / info about this happening and make the user aware that one was chosen over the other.

Also, we need to have tests considering this scenario.  Very nice catch from Xynnn!

Reply:
Let me take this back, I didn't initially got that the only way the configuration file is passed via `-c config_file` is when using peer pods, as the kernel parameters wouldn't be present in the VM.  This code doesn't conflict with what we currently have and my comment above **does not apply**!

Maybe that was exactly the realisation that Xynnn had and the reason the comment was deleted :-)